### ISSUE TITLE
Support JSON format for start script

### ISSUE THREAD

**Issue by Jazcash** [beyond-all-reason/spring#1088]
Currently the launching script.txt file is a custom format. For various existing and future systems, being able to easily read and write this file would help a lot. Almost every language now has support for easiling reading/writing JSON out the box, so I think this would make the most sense.

**Comment by sprunk** [comment]
If somebody were to add a jason library to the engine it might also be a good opportunity to do the jason half of #877.

**Comment by p2004a** [comment]
There is already JSON library available in the engine, this one https://github.com/open-source-parsers/jsoncpp

**Comment by lhog** [comment]
Yes, JSON is used there in the engine quite extensively, that's why I tagged this one as a good first issue. Adding a JSON parser to the config is just a matter of writing some boilerplate code.

**Comment by gabrsar** [comment]
I'm interested on that task. My C++ is very rusty, but I got things quick.

**Comment by sprunk** [comment]
@gabrsar any recent advances?

**Comment by SlashScreen** [comment]
I would like to spearhead work on this. 

The new system would work like the old one, using the same options, but instead use a json file with a $schema that linters can use to enforce the data shape.

Recoil cannot parse JSON natively, but thankfully there are [single-header JSON parsers](https://github.com/sheredom/json.h). By pulling in a library like this, we don't have to build a terrible parser ourselves, and we have access to error messages we can forward to the info log when it fails.

We can then also have a $schema hosted here on Github or wherever that can inform the user on the proper shape of the JSON file, providing much-needed documentation.

It doesn't need to be JSON, it can be any format that can have an attached schema (TOML could work too).

I can start working on this if we come to a consensus on how this should be done. It seems pretty simple, since it would end up with the same startup options structure, just with a different file type.

Sorry for the harsh tone of this issue, I have had a lot of trouble with start scripts this afternoon, and right now if I have to keep dealing with this I am going to end up on the news.

**Comment by p2004a** [comment]
@SlashScreen see https://github.com/beyond-all-reason/tachyon/blob/master/schema/autohost/start/request.json that does exactly that: defines json format for starscript, it might be close to copy paste with some small changes.

For our purposes it's [serialized to `.txt` format](https://github.com/beyond-all-reason/recoil-autohost/blob/main/src/startScriptGen.ts) with [tdf serializer](https://github.com/p2004a/recoil-tdf).

**Comment by SlashScreen** [comment]
I see, so instead of serializing to the regular format we cut out the middle man.

**Comment by SlashScreen** [comment]
How is this used in this fork of the engine?

**Comment by p2004a** [comment]
> How is this used in this fork of the engine?

What I linked is not used *in* the engine. I've only linked you to an existing JSON schema that represents almost full startscript so that you don't need to create that from scratch.

**Comment by SlashScreen** [comment]
Ah, ok, thanks. Is there still a JSON library in the engine somewhere? I don't see any references to it.

**Comment by p2004a** [comment]
> I don't see any references to it.

https://github.com/search?q=repo%3Abeyond-all-reason%2FRecoilEngine+json.h&type=code

There is even two of them.

**Comment by SlashScreen** [comment]
My bad.
